### PR TITLE
adds support for bang methods which allow for checking on production …

### PR DESCRIPTION
**Pull request by jcarres-mdsol**
…environment

@cabbott  @jfeltesse-mdsol  @miwest929 

In many projects we are checking if we are in produciton and making some kind of adhoc solution to error when variables were not set. This takes that pattern into a gem for great merry and joy of its users.

It's backwards compatible as you can not set variables with ! in the terminal.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# 0.9.0
+
+* Feature: Adding a bang (!) at the end of method names, will raise when the variable
+ not found in production.
```
Comment (jfeltesse-mdsol) [inline CHANGELOG.md]: what about "**is** not found"?

**Comment by jfeltesse-mdsol** [comment]
Code wise it looks ok to me although there are bits I'm not sure I understand (like that `features/step_definitions/rails.rb` file) but I guess it's me needing to get serious about learning cucumber.

**Comment by jcarres-mdsol** [comment]
Build failed because can't download ruby. Reason probably is because S3 is down today

**Comment by HonoreDB** [comment]
Cool, makes sense to me. It'd be nice to make this be less Rails-specific (Sinatra has environments too!) maybe for now just extracting `in_production?` into its own method?

**Comment by jcarres-mdsol** [comment]
@HonoreDB  Very good suggestion, I hope 84c8908 satisfies it

**Comment by HonoreDB** [comment]
Cool, thank you! I'll merge at end of day today if nobody objects.

**Comment by jcarres-mdsol** [comment]
@HonoreDB  you forgot to merge :whale:

**Pull request merged**